src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1629,6 +1629,8 @@ PlanRouterQuery(Query *originalQuery, RelationRestrictionContext *restrictionCon
 				List **placementList, uint64 *anchorShardId, List **relationShardList,
 				bool replacePrunedQueryWithDummy, bool *multiShardModifyQuery)
 {
+	static int zeroShardQueryRoundRobin = 0;
```

Comment:
Can we have overflow problem if the server lives long enough?

Reply:
Yeah, I think this at least should be unsigned so `%` doesn't return negative index when overflow happens.

Reply:
Yep, good catch! Fixed it.